### PR TITLE
Gulp: port the `all`, `extension` and `jsdoc` targets

### PR DESCRIPTION
**Pull request by timvandermeij**
This requires the `run-sequence` dependency because Gulp does not have a way to run sequences of tasks inside a Gulp task. Gulp 4.0 will have support for this, but until that is released this is the recommended way to implement it.

This is a part of #7062.

### CODE REVIEW

**Comment by Snuffleupagus** [comment]
Looks good, thanks for the patch!

**Comment by Snuffleupagus** [comment]
Looking at the [IRC logs](http://logs.glob.uno/?c=pdfjs) it seems that neither the `gh-pages` branch, nor the `pdfjs-dist` repo, was updated after this was pushed.
Could this patch have broken somehow have broken something on the bots? @yurydelendik When you have time, can you please investigate what's happened?

**Comment by yurydelendik** [comment]
Logs are at http://107.22.172.223:8877/418831005ec78d0/output.txt

File: gulpfile.js [hunk 1]
```diff
@@ -339,6 +342,13 @@ gulp.task('default', function() {
   });
 });
 
+gulp.task('extension', function (done) {
+  console.log();
+  console.log('### Building extensions');
+
+  runSequence('locale', ['firefox', 'chromium'], done);
```
Comment (Snuffleupagus) [inline gulpfile.js]: According to https://www.npmjs.com/package/run-sequence, an Array means parallel execution. Should we just change that and see if it helps?

Comment (yurydelendik) [inline gulpfile.js]: See 85db14652531a4c234dd0239e530c3e0d1062176

Comment (yurydelendik) [inline gulpfile.js]: Logs at http://107.22.172.223:8877/50dec1e848d3a50/output.txt

**Comment by yurydelendik** [comment]
Different idea: based on `Preparing request for ./build/gh-pages/extensions/firefox/pdf.js.xpi
Unable to parse XPI: invalid return code from unzip`, the [node make web](https://github.com/mozilla/botio-files-pdfjs/blob/master/on_push.js#L19) continue it's execution while running gulp tasks.

**Comment by Snuffleupagus** [comment]
> Different idea: based on Preparing request for ./build/gh-pages/extensions/firefox/pdf.js.xpi Unable to parse XPI: invalid return code from unzip, the node make web continue it's execution while running gulp tasks.

Yeah, this seems basically like issue #7083 (and PR #7084) all over again :-(
We really need to fix that properly this time, if we want to transition from `make` to `gulp`.

**Comment by yurydelendik** [comment]
Fixed the issue by 1369997bbfe71e05a6f45aee303a3d5f8d312798 -- we exited after any gulp execution.

**Pull request merged**